### DISCUSSION TITLE
Snap Lens Tools 403 issue

### DISCUSSION THREAD

**Discussion by TheeGuped** [ptrumpis/snap-camera-server#174] [Q&A] [answered]
hii, im having issues with the snap lens tools, i keep getting the same 403 error 
![image](https://github.com/user-attachments/assets/b7cf94d3-098a-4ac4-899d-d964f4a4dbef)
how can i fix it?

**Comment by alexirUDC** [comment]
I'm having the same issue, I don't know if it's only with new lens or if the web changed something

**Comment by dcordova20** [comment, marked as answer]
cant take credit for this, i copied it from a discussion in the github for the snap lens tool. it worked for me though. 

What happens there is the 'fake browser' that the web crawler pretends to be doesn't look authentic enough to the host.

Go to the file ..\src\core\lens_crawler.py and modify line 36 as follows:
BEFORE:
response = requests.get(url)
AFTER:
def_agent = 'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:70.0) Gecko/20100101 Firefox/70.0'
header = {'user-agent': def_agent, 'referer': url}
response = requests.get(url,headers=header)

That should do it.

**Reply by DKogut17** [reply]
Sadly this doesn't work in October 2025

**Reply by dcordova20** [reply]
I would imagine not, unless you already have the .lns file from the lens, the snap lens tool isnt gonna do you any good anyway.  Snapchat put the kibosh on that quite a while back.
